scripts/utils.sh

```diff
@@ -31,3 +31,71 @@ function check_exit_code {
         fatal_error "${fail_msg}"
     fi
 }
+
+# Reimplement 'mkdir -p' with reporting on where permissions break down
+function create_directory_structure() {
+  # Ensure we are given a single path argument
+  if [ $# -ne 1 ]; then
+    echo "Function requires a single (relative or absolute) path argument" >&2
```

Comment:
`Function` -> `create_directory_structure`

```suggestion
    echo "ERROR: create_directory_structure requires a single (relative or absolute) path argument" >&2